opm/grid/LookUpData.hh

```diff
@@ -261,131 +395,222 @@ protected:
+    return fieldProp[fieldPropIdx];
+}
+
+template<typename Grid, typename GridView>
+template<typename EntityType>
+double Opm::LookUpData<Grid,GridView>::fieldPropDoubleFromEntity(const FieldPropsManager& fieldPropsManager,
+                                                                 const std::string& propString,
+                                                                 const EntityType& elem) const
+{
+    const auto& fieldPropVec = fieldPropsManager.get_double(propString);
+    return this ->operator()(elem,fieldPropVec);
+}
+
+template<typename Grid, typename GridView>
+double Opm::LookUpData<Grid,GridView>::fieldPropDouble(const FieldPropsManager& fieldPropsManager,
+                                                       const std::string& propString,
+                                                       const int& elemIdx) const
+{
+    const auto& fieldPropVec = fieldPropsManager.get_double(propString);
+    return this ->operator()(elemIdx,fieldPropVec);
+}
+
+template<typename Grid, typename GridView>
+template<typename EntityType>
+int Opm::LookUpData<Grid,GridView>::fieldPropIntFromEntity(const FieldPropsManager& fieldPropsManager,
+                                                           const std::string& propString,
+                                                           const EntityType& elem) const
+{
+    const auto& fieldPropVec = fieldPropsManager.get_int(propString);
+    return this ->operator()(elem,fieldPropVec);
+}
+
+template<typename Grid, typename GridView>
+int Opm::LookUpData<Grid,GridView>::fieldPropInt(const FieldPropsManager& fieldPropsManager,
+                                                 const std::string& propString,
+                                                 const int& elemIdx) const
+{
+    const auto& fieldPropVec = fieldPropsManager.get_int(propString);
+    return this ->operator()(elemIdx,fieldPropVec);
 }
 
 template<typename Grid, typename GridView>
 template<typename EntityType, typename GridType>
 typename std::enable_if_t<!std::is_same_v<GridType,Dune::CpGrid>,int>
-Opm::LookUpData<Grid,GridView>::getOriginIndexFromEntity(const EntityType& elem) const
+Opm::LookUpData<Grid,GridView>::getFieldPropIdxFromEntity(const EntityType& elem) const
 {
     static_assert(std::is_same_v<Grid,GridType>);
+    assert(elem.level() == 0); // LGRs (level>0) only supported for CpGrid.
     return this-> elemMapper_.index(elem);
 }
 
 template<typename Grid, typename GridView>
 template<typename EntityType, typename GridType>
 typename std::enable_if_t<std::is_same_v<GridType,Dune::CpGrid>,int>
-Opm::LookUpData<Grid,GridView>::getOriginIndexFromEntity(const EntityType& elem) const
+Opm::LookUpData<Grid,GridView>::getFieldPropIdxFromEntity(const EntityType& elem) const
 {
     static_assert(std::is_same_v<Grid,GridType>);
-    return elem.getOrigin().index();
+    if (isFieldPropInLgr_ && elem.level()) { // level > 0 == true ; level == 0 == false
```

Comment:
One question because of the `elemlevel()` check: Wouldn't `elem.getLgrElem()` also do the correct thing if the entity is not refined?

Reply:
the answer seems to be no,